src/ripple/app/consensus/RCLConsensus.h

```diff
@@ -63,9 +63,8 @@ class RCLConsensus
         InboundTransactions& inboundTransactions_;
         beast::Journal const j_;
 
-        NodeID const nodeID_;
-        PublicKey const valPublic_;
-        SecretKey const valSecret_;
+        // If the server is validating, the necessary keying information:
+        ValidatorKeys const& validatorKeys_;
```

Comment:
The advantage of the old code is we don't have to worry about lifetime issues. I code is correct - the value that is passed in lives in the application object, which should outlive this. Still, it makes the code a little more fragile.

I probably weakly prefer the old code, but fine as-is as well.

Reply:
The old code makes it more difficult to 're-key' a server (e.g. by allowing it to switch its manifest) although that's not something currently supported. It also dumps keying material in whatever random memory location the `this` exists in which makes it more difficult to "secure" the memory for sensitive keying material (e.g. by marking it as "do not include in core dump" and by other similar, high-security techniques, including, but not limited do storing the key itself inside an HSM).

I understand the concern here, but the code is structured so that the object being passed in a singleton. I guess we could _reinforce_ that by passing a `std::shared_ptr` or somesuch.